cross/ntopng/Makefile

```diff
@@ -0,0 +1,86 @@
+LICENSE  = GPLv3
+
+include ../../mk/spksrc.common.mk
+
+SYNOLOGY_OS = DSM
+ifeq ($(findstring $(ARCH), $(SRM_ARCHS)),$(ARCH))
+SYNOLOGY_OS = SRM
+endif
+
+# similar to 'uname -m' on the target plattform (not essential, for information only)
+PACKAGE_MACHINE = x86_64
+
+ifeq ($(findstring $(ARCH), $(ARM5_ARCHES)),$(ARCH))
+PACKAGE_MACHINE = armv5
+endif
+ifeq ($(findstring $(ARCH), $(ARM7_ARCHES)),$(ARCH))
+PACKAGE_MACHINE = armv7l
+endif
+ifeq ($(findstring $(ARCH), $(ARM8_ARCHES)),$(ARCH))
+PACKAGE_MACHINE = aarch64
+endif
+ifeq ($(findstring $(ARCH), $(PPC_ARCHES)),$(ARCH))
+PACKAGE_MACHINE = ppc
+endif
+ifeq ($(findstring $(ARCH), $(x86_ARCHES)),$(ARCH))
+PACKAGE_MACHINE = i686
+endif
+
+GNU_CONFIGURE = 1
+PRE_CONFIGURE_TARGET = ntopng_pre_configure_target
+POST_CONFIGURE_TARGET = ntopng_post_configure_target
+PRE_COMPILE_TARGET = ntopng_pre_compile_target
+
+CONFIGURE_ARGS  = --prefix=$(INSTALL_PREFIX)
+CONFIGURE_ARGS += --with-ndpi-includes=$(STAGING_INSTALL_PREFIX)/include/ndpi
+CONFIGURE_ARGS += --with-ndpi-static-lib=$(STAGING_INSTALL_PREFIX)/lib
+
+# let ./configure find mysql_config (it is a script and works for cross compile)
+ENV += "PATH=$(PATH):$(STAGING_INSTALL_PREFIX)/bin"
+ENV += DESTDIR=$(INSTALL_DIR)
+
+
+include ../../mk/spksrc.cross-cc.mk
+
+.PHONY: ntopng_pre_configure_target
+ntopng_pre_configure_target:
+	@$(RUN) ./autogen.sh
+	@$(MSG) "- patch configure file to define synology specific package info"
+	@$(RUN) sed -i 's|#define PACKAGE_OSNAME "$${OSNAME}"|#define PACKAGE_OSNAME "Synology $(SYNOLOGY_OS) $(TCVERSION)+"|g' ./configure
+	@$(RUN) sed -i 's|#define PACKAGE_OS "$${OS}"|#define PACKAGE_OS "$(TC_ARCH)"|g' ./configure
+	@$(RUN) sed -i 's|#define PACKAGE_MACHINE "$$MACHINE"|#define PACKAGE_MACHINE "$(PACKAGE_MACHINE)"|g' ./configure
+	@$(MSG) "- patch sources to define real version instead of 'major.minor.currentdate', fix github links, reformat copyright date"
+	@$(RUN) sed -i "s|PACKAGE_VERSION='.*'|PACKAGE_VERSION='$(PKG_VERS)'|g" ./configure
+	@$(RUN) sed -i "s|PACKAGE_STRING='ntopng .*'|PACKAGE_STRING='ntopng $(PKG_VERS)'|g" ./configure
+	@$(RUN) sed -i 's|GIT_RELEASE=".*"|GIT_RELEASE="$(PKG_VERS)"|g' ./configure
+	@$(RUN) sed -i 's|"https://github.com/ntop/ntopng/commit/"..|"https://github.com/ntop/ntopng/releases/tag/"..|g' ./scripts/lua/runtime.lua
+	@$(RUN) sed -i 's|"https://github.com/ntop/ntopng/commit/"..|"https://github.com/ntop/ntopng/releases/tag/"..|g' ./scripts/lua/about.lua
+	@$(RUN) sed -i 's|"http://www.d3js.org|"http://d3js.org|g' ./scripts/lua/about.lua
+	@$(RUN) sed -i 's|"(C) 1998-\(..\) ntop.org"|"(C) 1998-20\1 ntop.org"|g' ./include/ntop_defines.h
+	@$(RUN) sed -i 's|copy; 1998-\(..\) - ntop.org"|copy; 1998-20\1 - ntop.org"|g' ./src/LuaEngine.cpp
+	@$(RUN) sed -i 's|"Welcome to %s %s v.%s - (C) 1998-\(..\) ntop.org"|"Welcome to %s %s v.%s - (C) 1998-20\1 ntop.org"|g' ./src/Ntop.cpp
```

Comment:
Please use patches from build framework. From my point of view, more readable and easier to maintain than such a set of sed commands, moreover when commands process same file more than once.

Reply:
It is not possible here to use patch files
1. the configure file is created with autogen.sh as pre_configure and is not available in the original sources
2. until now it is not available to use arch specific patches, and I will not recommend arch specific patch files with the sole reason to put the arch into a source file.
3. others are more cosmetic (like version and copyright date formating) and with regular expressions these are independent of source updates

Therefore I would not alter this